runtime/vm/createramclass.cpp

```diff
@@ -2142,6 +2142,30 @@ trcModulesSettingPackage(J9VMThread *vmThread, J9Class *ramClass, J9ClassLoader
 		}
 	}
 }
+static void
+initializeClassLinks(J9Class *ramClass, J9Class *superclass, J9MemorySegment *segment, UDATA options) {
+	ramClass->nextClassInSegment = *(J9Class **) segment->heapBase;
+	*(J9Class **)segment->heapBase = ramClass;
+
+	if (superclass == NULL) {
+		ramClass->subclassTraversalLink = ramClass;
+		ramClass->subclassTraversalReverseLink = ramClass;
+	} else {
+
+		if ((options & J9_FINDCLASS_FLAG_NO_SUBCLASS_LINK) == 0) {
+			J9Class* nextLink = superclass->subclassTraversalLink;
+
+			ramClass->subclassTraversalLink = nextLink;
+			nextLink->subclassTraversalReverseLink = ramClass;
+			superclass->subclassTraversalLink = ramClass;
+			ramClass->subclassTraversalReverseLink = superclass;
+		} else {
+			ramClass->subclassTraversalLink = ramClass;
+			ramClass->subclassTraversalReverseLink = ramClass;
+		}
```

Comment:
Since we hold the mutex over this operation, we can assign the ramClass to `subclassTraversalLink` & `subclassTraversalReverseLink` always, and overwrite in the one case:

```suggestion
	ramClass->subclassTraversalLink = ramClass;
	ramClass->subclassTraversalReverseLink = ramClass;
	if ((options & J9_FINDCLASS_FLAG_NO_SUBCLASS_LINK) == 0) {
		J9Class* nextLink = superclass->subclassTraversalLink;

		ramClass->subclassTraversalLink = nextLink;
		nextLink->subclassTraversalReverseLink = ramClass;
		superclass->subclassTraversalLink = ramClass;
		ramClass->subclassTraversalReverseLink = superclass;
	}
```

Reply:
This won't work if superclass is NULL. I agree about assigning the defaults first - just need to add the NULL check in the if.